core/src/sigverify.rs

```diff
@@ -119,22 +155,24 @@ pub fn generate_offsets(batches: &[Packets], recycler: &Recycler<TxOffset>) -> R
 
             let (sig_len, sig_start, msg_start_offset, pubkey_offset) =
                 get_packet_offsets(packet, current_offset);
-            let mut pubkey_offset = pubkey_offset;
 
             sig_lens.push(sig_len);
 
             trace!("pubkey_offset: {}", pubkey_offset);
-            let mut sig_offset = sig_start;
-            for _ in 0..sig_len {
-                signature_offsets.push(sig_offset);
-                sig_offset += size_of::<Signature>() as u32;
+            if sig_len > 0 {
```

Comment:
As mentioned in the pull request description, this part changes the runtime semantics. I'm assuming there will be no transaction with zero signature...